### ISSUE TITLE
Webtrees installation -- database 

### ISSUE THREAD

**Issue by mdavidsn** [fisharebest/webtrees#2091]
I received this error message after attempting to set up webtrees.  Can you help?

An unexpected database error occurred.

SQLSTATE[HY000]: General error: 1005 Can't create table `davidsn_webtrees`.`wt_user_setting` (errno: 121 "Duplicate key on write or update")
The webtrees developers would be very interested to learn about this error. If you contact them, they will help you resolve the problem.

**Comment by fisharebest** [comment]
What version of webtrees?

What version of MySQL (or MariaDB)?

Is this your only webtrees installation in this database?

**Comment by fisharebest** [comment]
> I was able to install webtrees on the same server and tried to transfer data but got an error message.

It may be have been easier to fix that error....

> SQLSTATE[HY000]: General error: 1005 Can't create table davidsn_webtrees.wt_user_setting (errno: 121 "Duplicate key on write or update")

This error message usually indicates a duplicate constraint name.

> I believe 1.4.6

You are installing webtrees 1.4.6???  webtrees 1.4.6 is *very* old.  The only reason to use it would be that it is the last version of webtrees to run on PHP 5.2.  Are you also using PHP 5.2?

Now, the setup script in 1.4.6 does not specify any constraint names for the table user_setting.  So, it is hard to see how it could cause this error.

Can you give me the sequence of events.  You started with a PGV installation.  Then what....

**Comment by mdavidsn** [comment]
Is there a file that tells me which version I am using?

**Comment by fisharebest** [comment]
The file includes/session.php should contain a line like this:

```
define('WT_VERSION',      '1.4.6');
```

**Comment by mdavidsn** [comment]
Yes, it is 1.4.6.  I believe that I began to use this version because I was
getting a php error.  I dont recall the specific error.  I am using php 5.6
although I belieev that I could change settings to use php 7.

**Comment by mdavidsn** [comment]
I should add that phpmyadmin hows that I have 2 webtrees dbs now - one
called webtrees and one called davidsn_webtrees.  Both have 4 tables with
no data.

**Comment by fisharebest** [comment]
>  I am using php 5.6

Then you should be able to use the latest version of webtrees - 1.7.12

> I dont recall the specific error.

Maybe try it again?

I didn't ask which version of PhpGedView you are using.  Older versions stored configuration in PHP files.  Newer versions stored configuration in the database.  Transferring from the newer versions is more thorroughly tested and reliable.  You might want to upgrade to the lastest PGV first....

> I should add that phpmyadmin hows that I have 2 webtrees dbs now - one called webtrees and one called davidsn_webtrees. Both have 4 tables with no data.

I would delete both of these before starting again.

BTW - how much "non genealogy" data do you have in PGV?  i.e. things like user accounts, "news" articles on the home pages, etc.?  If the answer is very little, then you may find it simpler to simply export a GEDCOM from PGV and then import this into a clean installation of webtrees.

**Comment by mdavidsn** [comment]
you wanted the sequence of events.  i have been slogging through this in my
spare time over the last week or two so i have this wrong, but here goes.

1.  Tried to install latest webtrees on synology.  got php error in setup
process and was directed by your forum to use the older version of webtrees
to get around problem.

2.  Installed older version and ran into setup problems relating to
appropriate db server credentials.  "SQLSTATE[HY000] [2002] No such file or
directory"

3.  Finally isolated error to use of 'localhost" server and port off by 1
(as indicated in Maria DB) and found that server 127.0.0.1 would allow
setup to proceed.

4.  I put in "webtrees" as db name (second time I tried with
davidsn_webtrees) and continue

5.  Fill in admn account data and continue

6.  get error SQLSTATE[HY000]: General error: 1005 Can't create table
`webtrees`.`wt_user_setting` (errno: 121 "Duplicate key on write or
update")

**Comment by mdavidsn** [comment]
I am using v phpgedview v. 4.2.4.  i will try reinstalling with more recent
version of webtrees.  If I run into the same php problem I will get back to
you.  I think your suggestion about simply importing a gedcom makes sense.
I dont have a huge number of imported docs.  thanks

**Comment by mdavidsn** [comment]
Greg, I removed the old version and uploaded the new version.  I had gotten
to the screen asking for my name, username and password.  When I hit
continue, I now get this error message:

Fatal error: Class 'Fisharebest\Webtrees\Theme\FabTheme' not found in
/hermes/walnacweb01/walnacweb01ab/b2419/pow.davidson2/htdocs/webtrees/themes/fab/theme.php
on line 20

Any thoughts on this? Mark

**Comment by fisharebest** [comment]
> Class 'Fisharebest\Webtrees\Theme\FabTheme' not found

Do you have the file `app/Theme/FabTheme.php`?
Do the contents look like https://github.com/fisharebest/webtrees/blob/1.7.12/app/Theme/FabTheme.php

My guess is that you missed one or more files when uploading to your server.

**Comment by fisharebest** [comment]
Close due to inactivity.  If you still have problems, please create a new issue.
